### PR TITLE
feat: add Imagen watermarking and captions samples and tests

### PR DESCRIPTION
**Pull request by irataxy**
## Description

Fixes b:355484599

- [X] **Tests** pass:   `mvn clean verify` **required**
- [X] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: aiplatform/src/main/java/aiplatform/imagen/VerifyImageWatermarkSample.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+      for (Value prediction : predictResponse.getPredictionsList()) {
+        Map<String, Value> fieldsMap = prediction.getStructValue().getFieldsMap();
+        if (fieldsMap.containsKey("decision")) {
+          System.out.format(
+              "Watermark verification result: %s", fieldsMap.get("decision").getStringValue());
+          // "ACCEPT" if the image contains a digital watermark
+          // "REJECT" if the image does not contain a digital watermark
+        }
+      } // only 1 prediction returned
```
Comment (Sita04) [inline aiplatform/src/main/java/aiplatform/imagen/VerifyImageWatermarkSample.java:95]: Please remove this dangling comment.

File: aiplatform/src/main/java/aiplatform/imagen/VerifyImageWatermarkSample.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          predictionServiceClient.predict(
+              endpointName, Collections.singletonList(instances), parameters);
+
+      for (Value prediction : predictResponse.getPredictionsList()) {
+        Map<String, Value> fieldsMap = prediction.getStructValue().getFieldsMap();
+        if (fieldsMap.containsKey("decision")) {
+          System.out.format(
+              "Watermark verification result: %s", fieldsMap.get("decision").getStringValue());
+          // "ACCEPT" if the image contains a digital watermark
+          // "REJECT" if the image does not contain a digital watermark
```
Comment (Sita04) [inline aiplatform/src/main/java/aiplatform/imagen/VerifyImageWatermarkSample.java:93]:
```suggestion
          // "ACCEPT" if the image contains a digital watermark
          // "REJECT" if the image does not contain a digital watermark
          System.out.format(
              "Watermark verification result: %s", fieldsMap.get("decision").getStringValue());
```

**Review by Sita04 — APPROVED** [review]
Couple nits. Please address those before merging.

File: aiplatform/src/main/java/aiplatform/imagen/VerifyImageWatermarkSample.java (hunk 1 above)
Comment (irataxy) [reply aiplatform/src/main/java/aiplatform/imagen/VerifyImageWatermarkSample.java]: Removed

**Pull request merged**